packages/typegpu/src/core/function/tgpuComputeFn.ts

```diff
@@ -55,28 +55,30 @@ export interface ComputeFnOptions {
  * Creates a shell of a typed entry function for the compute shader stage. Any function
  * that implements this shell can perform general-purpose computation.
  *
- * @param workgroupSize
+ * @param options.in
+ *   Record with builtins used by the compute shader.
+ * @param options.workgroupSize
  *   Size of blocks that the thread grid will be divided into (up to 3 dimensions).
  */
-export function computeFn<ComputeIn extends Record<string, AnyComputeBuiltin>>(
-  argTypes: ComputeIn,
-  options: ComputeFnOptions,
-): TgpuComputeFnShell<ComputeIn> {
-  const { workgroupSize } = options;
-
+export function computeFn<
+  ComputeIn extends Record<string, AnyComputeBuiltin>,
+>(options: {
+  in: ComputeIn;
+  workgroupSize: number[];
+}): TgpuComputeFnShell<ComputeIn> {
```

Comment:
We should allow `in` to be omitted when there's no input to the function. We can achieve that with function overloading.